utils/uri/matcher.go

```diff
@@ -0,0 +1,75 @@
+package uri
+
+import (
+	"errors"
+	"strings"
+)
+
+var (
+	ErrInvalidURI = errors.New("invalid URI string")
+)
```

Comment:
This could probably just be `var ErrInvalidURI = errors.New("invalid URI string")`

Reply:
Yep, I wasn't sure if I was going to add more errors, I think I can clean it up now though